File: swagger/models/client_index.py
from .base_model import BaseModel


class ClientIndex(BaseModel):
	"""
	A client index.

	Attributes:

		id
		The unique ID of the client index.

		name
		The name of the client index.

		required_business_mode
		When `true`, indicates that the index is required when creating
		profiles in business mode.

		required_consumer_mode
		When `true`, indicates that the index is required when creating
		profiles in consumer mode.

		values
		Contains information about the index's possible values.

		action
		The action performed on this object.
	"""

	input_map = {
		'Id': 'id',
		'Name': 'name',
		'RequiredBusinessMode': 'required_business_mode',
		'RequiredConsumerMode': 'required_consumer_mode',
		'Values': 'values',
		'Action': 'action',
		}

	output_map = {
		'id': 'Id',
		'name': 'Name',
		'required_business_mode': 'RequiredBusinessMode',
		'required_consumer_mode': 'RequiredConsumerMode',
		'values': 'Values',
		'action': 'Action',
		}

File: swagger/Models/ClientIndex.php
<?php

namespace App\MindBody\Models;

/**
 * A client index.
 */
class ClientIndex extends BaseModel
{
	/**
	 * The unique ID of the client index.
	 * 
	 * @var integer format:int32
	 */
	public $id;
	/**
	 * The name of the client index.
	 * 
	 * @var string 
	 */
	public $name;
	/**
	 * When `true`, indicates that the index is required when creating
	 * profiles in business mode.
	 * 
	 * @var boolean 
	 */
	public $requiredBusinessMode;
	/**
	 * When `true`, indicates that the index is required when creating
	 * profiles in consumer mode.
	 * 
	 * @var boolean 
	 */
	public $requiredConsumerMode;
	/**
	 * Contains information about the index's possible values.
	 * 
	 * @var ClientIndexValue[] 
	 */
	public $values;
	/**
	 * The action performed on this object.
	 * 
	 * @var string 
	 */
	public $action;

	protected function getInputMap()
	{
		return [
			'Id' => ['id', 'integer', 'int32'],
			'Name' => ['name', 'string', null],
			'RequiredBusinessMode' => ['requiredBusinessMode', 'boolean', null],
			'RequiredConsumerMode' => ['requiredConsumerMode', 'boolean', null],
			'Values' => ['values', 'array', ClientIndexValue::class],
			'Action' => ['action', 'string', null],
			];
	}

	protected function getOutputMap()
	{
		return [
			'id' => ['Id', 'integer', 'int32'],
			'name' => ['Name', 'string', null],
			'requiredBusinessMode' => ['RequiredBusinessMode', 'boolean', null],
			'requiredConsumerMode' => ['RequiredConsumerMode', 'boolean', null],
			'values' => ['Values', 'array', ClientIndexValue::class],
			'action' => ['Action', 'string', null],
			];
	}
}
